cura/Settings/PerObjectContainerStack.py

```diff
@@ -0,0 +1,49 @@
+from typing import Any, Optional
+
+from UM.Decorators import override
+from UM.Logger import Logger
+from UM.Settings.Interfaces import PropertyEvaluationContext
+from UM.Settings.ContainerStack import ContainerStack
+
+
+class PerObjectContainerStack(ContainerStack):
+
+    @override(ContainerStack)
+    def getProperty(self, key: str, property_name: str, context: Optional[PropertyEvaluationContext] = None) -> Any:
+        if context is None:
+            context = PropertyEvaluationContext()
+        context.pushContainer(self)
+
+        # this import has to be here otherwise there will be a circular import loop
+        from cura.CuraApplication import CuraApplication
+        global_stack = CuraApplication.getInstance().getGlobalContainerStack()
```

Comment:
Try `Application.getInstance().getGlobalContainerStack()`. You won't have a circular import then.

Reply:
Fixed in b84500f742db7ffb9352065027024b0a3f913ec6